x-pack/plugins/security_solution/public/management/pages/policy/view/host_isolation_exceptions/components/assign_flyout.tsx

```diff
@@ -0,0 +1,303 @@
+            />
+          }
+        />
+      );
+    }
+  }, [
+    allPossibleExceptionsRequest.data?.total,
+    allPossibleExceptionsRequest.isLoading,
+    currentFilter,
+    exceptionsRequest.data?.total,
+    exceptionsRequest.isLoading,
+  ]);
+
+  return (
+    <EuiFlyout onClose={onClose} data-test-subj="hostIsolationExceptions-assign-flyout">
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="m">
+          <h2>
+            <FormattedMessage
+              id="xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.flyout.title"
+              defaultMessage="Assign host isolation exceptions"
+            />
+          </h2>
+        </EuiTitle>
+        <EuiSpacer size="m" />
+        <FormattedMessage
+          id="xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.flyout.subtitle"
+          defaultMessage="Select host isolation exceptions to add to {policyName}"
+          values={{ policyName: policy.name }}
+        />
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        {(exceptionsRequest.data?.total || 0) > MAX_ALLOWED_RESULTS ? searchWarningMessage : null}
+        <SearchExceptions
+          onSearch={handleOnSearch}
+          placeholder={i18n.translate(
+            'xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.searh.label',
+            {
+              defaultMessage: 'Search host isolation exceptions',
+            }
+          )}
+          hideRefreshButton
+        />
+        <EuiSpacer size="m" />
+
+        {!exceptionsRequest.isLoading ? (
+          <PolicyArtifactsAssignableList
+            data-test-subj="hostIsolationExceptions-assignable-list"
+            artifacts={exceptionsRequest.data}
+            selectedArtifactIds={selectedArtifactIds}
+            isListLoading={exceptionsRequest.isLoading}
+            selectedArtifactsUpdated={handleSelectArtifact}
+          />
+        ) : null}
+
+        {noItemsMessage}
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              data-test-subj="hostIsolationExceptions-assign-cancel-button"
+              onClick={onClose}
+            >
+              <FormattedMessage
+                id="xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.flyout.cancel"
+                defaultMessage="Cancel"
+              />
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              className="eui-textTruncate"
+              style={{ maxWidth: '300px' }}
+              data-test-subj="hostIsolationExceptions-assign-confirm-button"
+              fill
+              onClick={handleOnConfirmAction}
+              isLoading={exceptionsRequest.isLoading || mutation.isLoading}
+              disabled={isEmpty(selectedArtifactIds)}
+              title={policy.name}
```

Comment:
Did you tried this with long names?

Reply:
the EUI button already handles those cases 
![image](https://user-images.githubusercontent.com/227916/146914735-604fc481-6edb-4ae3-815e-7180ab4f5f81.png)